manipulation/src/functTest/java/org/jboss/gm/manipulation/EHCacheProjectFunctionalTest.java

```diff
@@ -0,0 +1,86 @@
+package org.jboss.gm.manipulation;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.gradle.testkit.runner.BuildResult;
+import org.gradle.testkit.runner.TaskOutcome;
+import org.gradle.util.GradleVersion;
+import org.jboss.gm.common.JVMTestSetup;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.RestoreSystemProperties;
+import org.junit.contrib.java.lang.system.SystemOutRule;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestRule;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.jboss.gm.common.JVMTestSetup.JDK8_DIR;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
+
+public class EHCacheProjectFunctionalTest {
+
+    @Rule
+    public final SystemOutRule systemOutRule = new SystemOutRule().enableLog().muteForSuccessfulTests();
+
+    @Rule
+    public TemporaryFolder tempDir = new TemporaryFolder();
+
+    @Rule
+    public final TestRule restoreSystemProperties = new RestoreSystemProperties();
+
+    @BeforeClass
+    public static void setupJVM() throws IOException {
+        JVMTestSetup.setupJVM();
+    }
+
+    @Test
+    public void ensurePublish()
+            throws IOException, URISyntaxException, GitAPIException {
+        assumeTrue(GradleVersion.current().compareTo(GradleVersion.version("7.0")) >= 0);
+
+        final File simpleProjectRoot = tempDir.newFolder("ehcache3");
+
+        try (Git ignored = Git.cloneRepository()
+                .setURI("https://github.com/ehcache/ehcache3.git")
+                .setDirectory(simpleProjectRoot)
+                .setBranch("refs/tags/v3.10.2")
+                .setDepth(1)
+                .call()) {
+            System.out.println("Cloned ehcache3 to " + simpleProjectRoot);
+        }
+
+        // Now copy in overrides to enable this subset of unit test to work:
+        //   Modify Maven repository
+        //   Modify docs generation
```

Comment:
The docs modification in clustered/ehcache-clustered is the only thing I haven't got working yet for ehcache.

Reply:
I've figured out now (and put details on NCLSUP-793) why docs is causing a problem. Due to them having a custom build plugin, it seems to conflict. I don't think this is something GME can handle but have suggested a solution for the product team using a groovy script.